code/game/mecha/combat/neovgre

```diff
@@ -0,0 +1,96 @@
+	internals_action.Grant(user, src)
+	cycle_action.Grant(user, src)
+	lights_action.Grant(user, src)
+	stats_action.Grant(user, src)
+	strafing_action.Grant(user, src)
+
+/obj/mecha/combat/neovgre/RemoveActions(mob/living/user, human_occupant = 0)
+	internals_action.Remove(user)
+	cycle_action.Remove(user)
+	lights_action.Remove(user)
+	stats_action.Remove(user)
+	strafing_action.Remove(user)
+
+/obj/mecha/combat/neovgre/MouseDrop_T(mob/M, mob/user)
+	if(!is_servant_of_ratvar(user))
+		to_chat(user, "<span class='brass'>BEGONE HERETIC!</span>")
+		return
+	else
+		..()
+
+/obj/mecha/combat/neovgre/moved_inside(mob/living/carbon/human/H)
+	var/list/Itemlist = H.get_contents()
+	for(var/obj/item/clockwork/slab/W in Itemlist)
+		to_chat(H, "<span class='brass'>You safely store [W] inside [src].</span>")
+		qdel(W)
+	. = ..()
+
+/obj/mecha/combat/neovgre/obj_destruction()
+	for(var/mob/M in src)
+		to_chat(M, "<span class='brass'>You are consumed by the fires raging within Neovgre...</span>")
+		M.dust()
+	playsound(src, 'sound/magic/lightning_chargeup.ogg', 100, 0)
+	src.visible_message("<span class = 'userdanger'>The reactor has gone critical, its going to blow!</span>")
+	addtimer(CALLBACK(src,.proc/go_critical),breach_time)
+
+/obj/mecha/combat/neovgre/proc/go_critical()
+	explosion(get_turf(loc), 3, 5, 10, 20, 30)
+	Destroy(src)
+
+/obj/mecha/combat/neovgre/container_resist(mob/living/user)
+	to_chat(user, "<span class='brass'>Neovgre requires a lifetime commitment friend, no backing out now!</span>")
+	return
+
+/obj/mecha/combat/neovgre/process()
+	..()
+	if(GLOB.ratvar_awakens) // At this point only timley intervention by lord singulo could hople to stop the superweapon
+		cell.charge = INFINITY
+		max_integrity = INFINITY
+		obj_integrity = max_integrity
+		CHECK_TICK //Just to be on the safe side lag wise
+	else if(cell.charge < cell.maxcharge)
+		for(var/obj/effect/clockwork/sigil/transmission/T in range(SIGIL_ACCESS_RANGE, src))
+			var/delta = min(recharge_rate, cell.maxcharge - cell.charge)
+			if (get_clockwork_power() <= delta)
+				cell.charge += delta
+				adjust_clockwork_power(-delta)
+			CHECK_TICK
+
+/obj/mecha/combat/neovgre/Initialize()
+	.=..()
+	GLOB.neovgre_exists ++
+	var/obj/item/mecha_parts/mecha_equipment/weapon/energy/laser/heavy/neovgre/N = new
+	N.attach(src)
+
+/obj/structure/mecha_wreckage/durand/neovgre
+	name = "\improper Neovgre wreckage?"
+	desc = "On closer inspection this looks like the wreck of a durand with some spraypainted cardboard duct taped to it!"
+
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/laser/heavy/neovgre
+	equip_cooldown = 8 //Rapid fire heavy laser cannon, simple yet elegant
+	energy_drain = 30
+	name = "Aribter Laser Cannon"
+	desc = "Please re-attach this to neovgre and stop asking questions about why it looks like a normal Nanotrasen issue Solaris laser cannon - Nezbere"
```

Comment:
wait, does this mean the occupant can detach the laser cannon from inside? if so, why have the coderspeak in the desc? god!

Reply:
the cannon cant be deattached normally, the neovgre doesnt have a status panel, or an eject one.